scripts/neighbor_advertiser

```diff
@@ -82,16 +84,34 @@ def log_error(msg):
 config_db = None
 
 
-#
-# Get switch info and intf addr
-#
-
 def connect_config_db():
     global config_db
     config_db = ConfigDBConnector()
     config_db.connect()
 
 
+#
+# Check if a DIP returned from ferret is in any of this switch's VLANs
+#
+
+is_dip_in_device_vlan = None
+
+
+def get_device_vlan_checker():
+    global is_dip_in_device_vlan
+
+    vlan_interfaces = config_db.get_table('VLAN_INTERFACE')
+
+    def check_device(ferret_server_ip_addr):
+        return any(IPAddress(ferret_server_ip_addr) in IPNetwork(vlan_int[1]) for vlan_int in vlan_interfaces.iterkeys())
```

Comment:
>vlan_int[1]) for vlan_int in vlan_interfaces.iterkeys [](start = 65, length = 53)

what if vlan_interface has ipv6 address? we should only filter ipv4 address here.

Reply:
we also have a case where vlan interface only contains interface name, but not IP address. we should check vlan_int[1] is available or not. you can sync with prince on this case.

---
In reply to: [328399686](https://github.com/Azure/sonic-utilities/pull/670#discussion_r328399686) [](ancestors = 328399686)

Reply:
Good catch!